server/util/retry/retry.go

```diff
@@ -12,21 +12,28 @@ type Options struct {
 	InitialBackoff time.Duration // How long to wait after the first request
 	MaxBackoff     time.Duration // Max amount of time to wait for a single request
 	Multiplier     float64       // Next backoff is this * previous backoff
-	MaxRetries     int           // Max number of retries; 0 is inf
+	MaxRetries     int           // Max number of retries; 0 is based on time
 }
 
 type Retry struct {
 	opts *Options
 	ctx  context.Context
 
 	currentAttempt int
+	maxAttempts    int
 	isReset        bool
 }
 
 func New(ctx context.Context, opts *Options) *Retry {
+	maxAttempts := opts.MaxRetries
+	if maxAttempts <= 0 {
+		maxAttempts = int(math.Ceil(math.Log(float64(opts.MaxBackoff)/float64(opts.InitialBackoff)) / math.Log(opts.Multiplier)))
```

Comment:
this seems difficult to modify down the road if we want to add things like random jitter to this util -- wdyt about directly detecting when we're at the max backoff instead?